packages/react-ui/src/app/builder/builder-hooks.ts

```diff
@@ -223,3 +224,67 @@ function getStateAtPath({
   });
   return targetMap;
 }
+
+const getAllStepsMentions: (state: BuilderState) => MentionTreeNode[] = (
+  state,
+) => {
+  const { selectedStep, flowVersion } = state;
+  if (!selectedStep || !flowVersion || !flowVersion.trigger) {
+    return [];
+  }
+  const step = flowHelper.getStep(flowVersion, selectedStep.stepName);
+  if (!step) {
+    return [];
+  }
+  const path = flowHelper.findPathToStep({
+    stepToFind: step,
+    trigger: flowVersion?.trigger,
+  });
+
+  return path.map((s) => {
+    const stepMentionNode: MentionTreeNode =
+      dataSelector.traverseStepOutputAndReturnMentionTree({
+        stepOutput: s.settings.inputUiInfo?.currentSelectedData,
+        propertyPath: s.name,
+        displayName: s.displayName,
+      });
+    const stepNeedsTesting =
+      s.settings.inputUiInfo?.currentSelectedData === undefined;
+    return {
+      ...stepMentionNode,
+      data: {
+        ...stepMentionNode.data,
+        displayName: `${s.dfsIndex}. ${s.displayName}`,
+      },
+      children: stepNeedsTesting
+        ? [
+            {
+              data: {
+                displayName: 'Testing Step',
+                propertyPath: s.name,
+                isTestStepNode: true,
+                isSlice: false,
+              },
+              key: s.name,
+            },
+          ]
+        : stepMentionNode.children,
+    };
+  });
```

Comment:
```suggestion
   const stepNeedsTesting =
      s.settings.inputUiInfo?.currentSelectedData === undefined;
    if (stepNeedsTesting) {
      return {
        key: s.name,
        data: {
          displayName: `${s.dfsIndex}. ${s.displayName}`,
        },
        children: [
          {
            key: s.name,
            data: {
              displayName: 'Testing Step',
              propertyPath: s.name,
              isTestStepNode: true,
              isSlice: false,
            },
            children: [],
          },
        ]
      };
    }
    const stepMentionNode: MentionTreeNode =
      dataSelector.traverseStepOutputAndReturnMentionTree({
        stepOutput: s.settings.inputUiInfo?.currentSelectedData,
        propertyPath: s.name,
        displayName: s.displayName,
      });
    return stepMentionNode;
```
Something like this, (This is not exact code, but it should give you an idea of what I am trying to do)